extensions/azurecore/src/azureResource/tree/treeProvider.ts

```diff
@@ -37,13 +37,16 @@ export class AzureResourceTreeProvider implements TreeDataProvider<TreeNode>, IA
 	private hookAccountService(appContext: AppContext): void {
 		this.accountService = appContext.getService<IAzureResourceAccountService>(AzureResourceServiceNames.accountService);
 		if (this.accountService) {
-			this.accountService.onDidChangeAccounts((e: azdata.DidChangeAccountsParams) => {
+			this.accountService.onDidChangeAccounts(async (e: azdata.DidChangeAccountsParams) => {
+				// This event sends it per provider, we need to make sure we get all the azure related accounts
+				let accounts = await this.accountService.getAccounts();
+				accounts = accounts.filter(a => a.key.providerId.startsWith('azure'));
```

Comment:
Not a fan of doing checks like this - why are you doing startsWith instead of just checking the providerId directly?

Reply:
Because there are multiple providers under `azure`.

Its why I changed the name to all start with `azure`. This is what was necessary to support non-public clouds. We could potentially have a `providerFamily`. But we could also achieve that by just using the string with a specific format.

Reply:
We can even check on the account provider registration for it to have a `providerFamily_providerName` format.

Reply:
Yeah, I dunno. It's probably fine like this but I just get unreasonably paranoid about some extension author coming in and registering a new provider that also starts with azure that breaks our stuff. But I suppose that's always a concern with extensions and people can always just uninstall it. 

No big concern here in the end so do whatever you feel is best 😄